pkg/api/handlers/compat/containers_create.go

```diff
@@ -69,7 +72,16 @@ func CreateContainer(w http.ResponseWriter, r *http.Request) {
 	}
 	body.Config.Image = imageName
 
-	newImage, resolvedName, err := runtime.LibimageRuntime().LookupImage(body.Config.Image, nil)
+	lookupImageOptions := libimage.LookupImageOptions{}
+	if query.Platform != "" {
+		var err error
+		lookupImageOptions.OS, lookupImageOptions.Architecture, lookupImageOptions.Variant, err = parse.Platform(query.Platform)
+		if err != nil {
+			utils.Error(w, 200, fmt.Errorf("parsing platform: %w", err))
```

Comment:
or wait, why return 200 with an error? whit should return a proper error status code

Reply:
Oh thanks for catching that!  It's a debug relict as I expected to see a bad-request but `parse.Platform` does not validate the parsed values.

Reply:
Updated